conans/test/integration/toolchains/meson/test_mesontoolchain.py

```diff
@@ -205,3 +206,80 @@ def test_clang_cl_vscrt(build_type, runtime, vscrt):
     t.run("install . -pr:h=profile -pr:b=profile")
     content = t.load(MesonToolchain.native_filename)
     assert f"b_vscrt = '{vscrt}'" in content
+
+
+def test_env_vars_from_build_require():
+    conanfile = textwrap.dedent("""
+    from conan import ConanFile
+    import os
+
+    class HelloConan(ConanFile):
+        name = 'hello_compiler'
+        version = '1.0'
+        def package_info(self):
+            self.buildenv_info.define("CC", "CC_VALUE")
+            self.buildenv_info.define("CC_LD", "CC_LD_VALUE")
+            self.buildenv_info.define("CXX", "CXX_VALUE")
+            self.buildenv_info.define("CXX_LD", "CXX_LD_VALUE")
+            self.buildenv_info.define("AR", "AR_VALUE")
+            self.buildenv_info.define("STRIP", "STRIP_VALUE")
+            self.buildenv_info.define("AS", "AS_VALUE")
+            self.buildenv_info.define("WINDRES", "WINDRES_VALUE")
+            self.buildenv_info.define("PKG_CONFIG", "PKG_CONFIG_VALUE")
+            self.buildenv_info.define("LD", "LD_VALUE")
+    """)
+    client = TestClient()
+    client.save({"conanfile.py": conanfile})
+    client.run("create .")
+
+    conanfile = textwrap.dedent("""
+    from conan import ConanFile
+    class HelloConan(ConanFile):
+        name = 'consumer'
+        version = '1.0'
+        generators = "MesonToolchain"
+        settings = "os", "arch", "compiler", "build_type"
+
+        def requirements(self):
+            self.tool_requires("hello_compiler/1.0", )
+    """)
+    # Now, let's check how all the build env variables are applied at consumer side
+    client.save({"conanfile.py": conanfile})
+    client.run("install . -pr:h=default -pr:b=default")
+    content = client.load("conan_meson_native.ini")
+    assert "c = 'CC_VALUE'" in content
+    assert "cpp = 'CXX_VALUE'" in content
+    assert "ld = 'LD_VALUE'" in content
+    assert "c_ld = 'CC_LD_VALUE'" in content
+    assert "cpp_ld = 'CXX_LD_VALUE'" in content
+    assert "ar = 'AR_VALUE'" in content
+    assert "strip = 'STRIP_VALUE'" in content
+    assert "as = 'AS_VALUE'" in content
+    assert "windres = 'WINDRES_VALUE'" in content
+    assert "pkgconfig = 'PKG_CONFIG_VALUE'" in content
+
+
+def test_check_c_cpp_ld_list_formats():
+    # Issue related: https://github.com/conan-io/conan/issues/14028
+    profile = textwrap.dedent("""
+       [settings]
+       os=Windows
+       arch=x86_64
+       compiler=gcc
+       compiler.version=9
+       compiler.cppstd=17
+       compiler.libcxx=libstdc++11
+       build_type=Release
+       [buildenv]
+       CC=aarch64-poky-linux-gcc  -mcpu=cortex-a53 -march=armv8-a+crc+crypto
```

Comment:
So Meson needs this defined as a list for the ``c``, ``cpp``, etc? This reads a bit ugly imho, I'd expect to have the compiler as one thing and the list of flags as cxxflags or similar

Reply:
I agree with that, but the thing is that you can pass all those arguments to the CC in one shot as a list and it seems to be working as it's said in the issue https://github.com/conan-io/conan/issues/14028.

Overall, we're not changing the behavior at all. We agree that this should not be the way to go, but we're only allowing the users to have those compiler and compiler flags together without changing those env variables (up to them). You can check the Meson documentation too (https://mesonbuild.com/Machine-files.html#binaries).

Reply:
@memsharded This is unforunately how Yocto configures the `CC` and similar environment variables.